snooty/parser.py

```diff
@@ -347,6 +347,29 @@ def handle_directive(
                 msg = f'Invalid "literalinclude": {err}'
                 self.diagnostics.append(Diagnostic.error(msg, util.get_line(node)))
             return
+        elif name == "include":
+            if argument_text is None:
+                self.diagnostics.append(
+                    Diagnostic.error(
+                        f'"{name}" expected a path argument', util.get_line(node)
+                    )
+                )
+                return
+
+            fileid, path = util.reroot_path(
+                Path(argument_text), self.docpath, self.source_path
+            )
+
+            # Validate if file exists
+            if not path.exists():
```

Comment:
`.is_file()` would be preferable to `.exists()`; the latter would permit a directory to be provided, for example